Runtime/Scripts/EdgeEventsManager.cs

```diff
@@ -609,17 +601,115 @@ IEnumerator UpdateLocation()
-        }
+        location = LocationService.RetrieveLocation();//throws location exception if location is not retrieved
         yield break;
       }
     }
-    #endregion
+
+    void ProcessLatency(Statistics stats)
+    {
+      Logger.Log("Processing Latency Received from Server");
+      if (stats.Avg > config.latencyThresholdTriggerMs)
+      {
+        Logger.Log("Current Latency (" + stats.Avg + ") > LatencyThreshold (" + config.latencyThresholdTriggerMs + ")");
+        Logger.Log("Performing FindCloudlet PerformanceMode");
+        PlatformIntegration pIntegration = new PlatformIntegration();
+        MatchingEngine matchingEngine = new MatchingEngine(pIntegration.CarrierInfo, pIntegration.NetInterface, pIntegration.UniqueID, pIntegration.DeviceInfo);
+        FCPerformanceThreadClass fcThreadObj = new FCPerformanceThreadClass
+          (matchingEngine: matchingEngine, sessionCookie: integration.matchingEngine.sessionCookie, location: location, hostOverride: hostOverride, portOverride: portOverride, callbackDelegate: FCCallback);
+        fcThreadObj.RunFCPerformance();
+      }
+      else
+      {
+        Logger.Log("Current Latency (" + stats.Avg + ") < LatencyThreshold (" + config.latencyThresholdTriggerMs + ")");
+      }
+      return;
+    }
+
+    // This callback is emitted from a ThreadPoolWorker
+    private void FCCallback(FindCloudletReply findCloudletReply)
+    {
+      processingStatus = LatencyProcessingStatus.Processed;
+      FCPerformanceReply = findCloudletReply;
+      if(fcThread != null)
+      {
+        fcThread.Interrupt();
+      }
+    }
+  }
+  #endregion
+
+  class FCPerformanceThreadClass
+  {
+    string hostOverride;
+    uint portOverride;
+    string sessionCookie;
+    MatchingEngine matchingEngine;
+    EdgeEventsManager.FCPerformanceCallback callback;
+    Loc location;
+    Thread FCPerformanceThread;
+
+    internal FCPerformanceThreadClass(MatchingEngine matchingEngine, string sessionCookie, Loc location, string hostOverride, uint portOverride,
+     EdgeEventsManager.FCPerformanceCallback callbackDelegate)
+    {
+      this.hostOverride = hostOverride;
+      this.portOverride = portOverride;
+      this.sessionCookie = sessionCookie;
+      this.matchingEngine = matchingEngine;
+      this.location = location;
+      callback = callbackDelegate;
+    }
+
+    internal void RunFCPerformance()
+    {
+      FCPerformanceThread = new Thread(new ThreadStart(RunFCPerformanceHelper));
+      try
+      {
+        FCPerformanceThread.Name = "FCPerformanceThread";
+        FCPerformanceThread.IsBackground = true;
+        FCPerformanceThread.Priority = System.Threading.ThreadPriority.Lowest;
+        FCPerformanceThread.Start();
+        Debug.Log("Waiting for FindCloudlet Performance Mode");
+        FCPerformanceThread.Join();
+      }
+      catch (ThreadInterruptedException tie)
+      {
+        callback(null);
+        FCPerformanceThread.Abort();
```

Comment:
Thread Abort should probably be used sparingly (Murder the thread in whatever state it's in, and cleanup after).